libvast/builtins/connectors/file.cpp

```diff
@@ -0,0 +1,262 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <vast/concept/parseable/to.hpp>
+#include <vast/concept/parseable/vast/data.hpp>
+#include <vast/detail/fdinbuf.hpp>
+#include <vast/detail/file_path_to_parser.hpp>
+#include <vast/detail/posix.hpp>
+#include <vast/detail/string.hpp>
+#include <vast/logger.hpp>
+#include <vast/plugin.hpp>
+
+#include <caf/error.hpp>
+
+#include <fcntl.h>
+#include <filesystem>
+#include <memory>
+#include <string_view>
+#include <unistd.h>
+
+namespace vast::plugins::file {
+
+class plugin : public virtual loader_plugin {
+public:
+  static constexpr inline auto max_chunk_size = size_t{16384};
+  static constexpr inline auto stdin_path = "-";
+
+  auto
+  make_loader(std::span<std::string const> args, operator_control_plane&) const
+    -> caf::expected<generator<chunk_ptr>> override {
+    auto read_timeout = read_timeout_;
+    auto path = std::string{};
+    auto file_type = std::filesystem::file_type::regular;
+    auto following = false;
+    for (auto i = size_t{0}; i < args.size(); ++i) {
+      const auto& arg = args[i];
+      VAST_TRACE("processing loader argument {}", arg);
+      if (arg == "--timeout") {
+        if (i + 1 == args.size()) {
+          return caf::make_error(ec::syntax_error,
+                                 fmt::format("missing duration value"));
+        }
+        if (auto parsed_duration = to<vast::duration>(args[i + 1])) {
+          read_timeout = std::chrono::duration_cast<std::chrono::milliseconds>(
+            *parsed_duration);
+          ++i;
+        } else {
+          return caf::make_error(ec::syntax_error,
+                                 fmt::format("could not parse duration: {}",
+                                             args[i + 1]));
+        }
+      } else if (arg == "--uds") {
+        file_type = std::filesystem::file_type::socket;
```

Comment:
Why can't we do this automatically by inspecting `std::filesystem::status`?

Reply:
We specified `--uds` options for explicit `file_type`/`is_socket` setting. 
I can implement an automatic file type inspection that can get overwritten by the `--uds` option.

Reply:
Inferring this automatically makes a lot of sense to me, no need to override it then or to parse the option in the first place.

Reply:
> Inferring this automatically makes a lot of sense to me, no need to override it then or to parse the option in the first place.

However, the `--uds` option that @Dakostu mentioned is also used in the `from uds` syntax. We would have to remove that, or use some other mechanism to communicate that that follows must be a socket.

Reply:
Can't we just remove the UDS loader in its entirety? We'd only need to distinguish in the file saver, where an option to write to a socket makes a lot of sense to have.

Reply:
Removing the UDS loader should be no problem.

Reply:
I removed it in https://github.com/tenzir/vast/pull/3088/commits/a3b046c402783025a72fa735ac583f17f6598ba3